### ISSUE TITLE
Captioning Novel Objects in Images  The Berkeley Artificial Intelligence Research Blog

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#1206]
Captioning Novel Objects in Images – The Berkeley Artificial Intelligence Research Blog<br>
<br>
https://ift.tt/2vLQIjt<br>
<br>
Daniel Seita<br>
<br>

<div>
<div>
  <h1>Captioning Novel Objects in Images</h1>
  <p>
  
  Subhashini Venugopalan and Lisa Anne Hendricks   
  
  
  Aug 8, 2017
  
  </p>
</div>

<article>
  <p>Given an image, humans can easily infer the salient entities in it, and describe the scene effectively, such as, where objects are located (in a forest or in a kitchen?), what attributes an object has (brown or white?), and, importantly, how objects interact with other objects in a scene (running in a field, or being held by a person etc.). The task of visual description aims to develop visual systems that generate contextual descriptions about objects in images. Visual description is challenging because it requires recognizing not only objects (bear), but other visual elements, such as actions (standing) and attributes (brown), and constructing a fluent sentence describing how objects, actions, and attributes are related in an image (such as the brown bear is standing on a rock in the forest).</p>

<h2 id="current-state-of-visual-description">Current State of Visual Description</h2>

<table>
  <tr>
    <td>
			<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/bear.png" alt="Brown bear in forest">
		</td>
    <td>
			<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/anteater.png" alt="Anteater in forest">
		</td>
  </tr>
  <tr>
    <td><p>
			<a href="http://jeffdonahue.com/lrcn/">LRCN</a> [Donahue et al. ‘15]: A brown bear standing on top of a lush green field. <br>
			<a href="http://captionbot.ai">MS CaptionBot</a> [Tran et al. ‘16]: A large brown bear walking through a forest.
		</p></td>
    <td><p>
			<a href="http://jeffdonahue.com/lrcn/">LRCN</a> [Donahue et al. ‘15]: A black <span>bear</span> is standing in the grass. <br>
			<a href="http://captionbot.ai">MS CaptionBot</a> [Tran et al. ‘16]: A <span>bear</span> that is eating some grass.
		</p></td>
  </tr>
</table>
<p><i>
	Descriptions generated by existing captioners on two images. On the left is an image of an object (bear) that is present in training data. On the right is an object (anteater) that the model hasn't seen in training.
</i></p>

<p>Current visual description or image captioning models work quite well, but they can only describe objects seen in existing image captioning training datasets, and they require a large number of training examples to generate good captions. To learn how to describe an object like “jackal” or “anteater” in context, most description models require many examples of jackal or anteater images with corresponding descriptions.  However, current visual description datasets, like <a href="https://bair.berkeley.edu/blog/2017/08/08/novel-object-captioning/mscoco.org">MSCOCO</a>, do not include descriptions about all objects. In contrast, recent works in object recognition through Convolutional Neural Networks (CNNs) can recognize hundreds of categories of objects. While object recognition models can recognize jackals and anteaters, description models cannot compose sentences to describe these animals correctly in context.  In our work, we overcome this problem by building visual description systems which can describe new objects without pairs of images and sentences about these objects.</p>

<h2 id="the-task-describing-novel-objects">The Task: Describing Novel Objects</h2>

<p>Here we define our task more formally.  Given a dataset consisting of pairs of images and descriptions (paired image-sentence data, e.g. <a href="https://bair.berkeley.edu/blog/2017/08/08/novel-object-captioning/mscoco.org">MSCOCO</a>) as well as images with object labels but no descriptions (unpaired image data, such as <a href="http://www.image-net.org/">ImageNet</a>) we wish to learn how to describe objects unseen in paired image-sentence data. To do this we must build a model which can recognize different visual constituents (e.g., jackal, brown, standing, and field) and compose these in novel ways to form a coherent description.  Below we describe the core components of our description model.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/image_0.png" alt="The novel visual description task">
</p>

<p>We aim to describe diverse objects which do not have training images with captions.</p>

<h3 id="using-external-sources-of-data">Using External Sources of Data</h3>

<p>In order to generate captions about diverse categories of objects outside the image-caption training data, we take advantage of external data sources. Specifically, we use ImageNet images with object labels as the unpaired image data source and sentences from unannotated text corpora such as Wikipedia as our text data source. These are used to train our visual recognition CNN and language model respectively.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/image_1.png" alt="Train effectively on external resources"><br>
<i>
Train effectively on external resources
</i>
</p>

<h3 id="capture-semantic-similarity">Capture semantic similarity</h3>

<p>We want to be able to describe unseen objects (e.g. from ImageNet) that are similar to objects that have been seen in the paired image-sentence training data. We use dense word embeddings to achieve this. Word embeddings are dense high dimensional representations of words where words with similar meaning are closer in the embedding space.</p>

<p>In our previous work, called “Deep Compositional Captioning (DCC)” [1] we first train a caption model on MSCOCO paired  image caption dataset. Then to describe novel objects, for each novel object (such as an okapi) we use word embeddings to identify an object that’s most similar amongst the objects in the MSCOCO dataset (in this case zebra). We then transfer (copy) the parameters learned by the model from the seen object to the unseen object (i.e. copy weights in the network corresponding to zebra to those corresponding to okapi).</p>

<h3 id="novel-object-captioning">Novel Object Captioning</h3>

<p>While the DCC model is able to describe several unseen object categories, copying parameters from one object to another can create sentences with grammatical artifacts. E.g. for the object ‘racket’ the model copies weights from ‘tennis’, which results in sentences such as “A man playing racket on court”. In our more recent work [2], we incorporate the embeddings directly within our language model. Specifically, we use <a href="https://nlp.stanford.edu/projects/glove/">GloVe embeddings</a> in the input and output of our language model. This implicitly enables the model to capture semantic similarity when describing unseen objects. This enables our model to generate sentences such as “A tennis player swinging a racket at a ball”. Additionally, incorporating the embeddings directly within the network makes our model end-to-end trainable.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/image_2.png" alt="Dense word embeddings to capture image similarity"><br>
<i>
Incorporate dense word embeddings in the language model to capture semantic similarity.
</i>
</p>

<h3 id="caption-model-and-forgetting-in-neural-networks">Caption model and forgetting in neural networks.</h3>

<p>We combine the outputs of the visual network and language model to a caption model. This model is similar to existing caption models which are also pre-trained on ImageNet. However, we observed that although the model is pre-trained on ImageNet, when the model is trained / tuned on the COCO image-caption dataset it tends to forget what it has seen before. The problem of forgetting in neural networks has also been observed by <a href="https://arxiv.org/abs/1312.6211">researchers at Montreal</a> as well as <a href="https://arxiv.org/abs/1612.00796">Google DeepMind</a> amongst others. In our work, we resolve this problem of forgetting using a joint training strategy.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/image_3.png" alt="Joint training to overcome forgetting"><br>
<i>
Share parameters and train jointly on different data/tasks to overcome "forgetting"
</i>
</p>

<p>Specifically, our network has three components: a visual recognition network, a caption model, and a language model. All three components share parameters and are jointly trained. During training, each batch of inputs contains some images with labels, a different set of images and captions, and some plain sentences. These three inputs train the different components of the network. Since the parameters are shared between the three components, the network is jointly trained to recognize objects in images, caption images and generate sentences. This joint training helps the network overcome the problem of forgetting, and enables the model to generate descriptions for many novel object categories.</p>

<h2 id="whats-next">What’s Next?</h2>

<p>One of the most common errors in our model comes from not recognizing objects, and one way to mitigate this is to use better visual features. Another common error comes from generating sentences which are not fluent (A cat and a cat on a bed) or may not appeal to “common sense”  (e.g. ‘A woman is playing gymnastics’ is not particularly correct since one doesn’t “play” gymnastics). It would be interesting to develop solutions that can overcome these issues.</p>

<p>While in this work, we proposes joint training as a strategy to overcome the problem of forgetting, it might not always be possible to train on lots of different tasks and datasets. A different way to approach the problem would be to build a model that can learn to compose descriptions based on visual information and object labels. Such a model should also be able to integrate objects on the fly i.e. currently we pre-train our model on a select set of objects, we should also think about how we can incrementally train our model on new data about some new concepts. Solving some of these problems can help develop better and more robust visual description models.</p>

<p>[<a href="https://vsubhashini.github.io/noc_examples.html">Links to more examples</a>]</p>

<p>[<a href="http://vsubhashini.github.io/noc.html#code">Links to trained models and code</a>]</p>

<h3 id="examples">Examples</h3>

<p>
<img src="http://bair.berkeley.edu/static/blog/novel_image_captioning/image_4.png" alt="Examples">
</p>

<hr>

<p><strong>This blog post is based on the following research papers:</strong></p>

<p>[1] L. A. Hendricks, S. Venugopalan, M. Rohrbach, R. Mooney, K. Saenko, and T. Darrell. Deep compositional captioning: Describing novel object categories without paired training data. In CVPR, 2016.</p>

<p>[2] S. Venugopalan, L. A. Hendricks, M. Rohrbach, R. Mooney, K. Saenko, and T. Darrell. Captioning images with diverse objects. In CVPR, 2017.</p>

</article>

</div><br>
<br>
<br>
<br>
<br>
<br>
via The Berkeley Artificial Intelligence Research Blog https://ift.tt/2EPVVfk<br>
<br>
October 24, 2018 at 02:21PM
